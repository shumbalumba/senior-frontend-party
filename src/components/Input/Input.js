// @flow

import * as React from 'react';
import styled from 'styled-components';

import { breakpoints } from '../../theme/media';

const InputWrapper = styled.div`
  position: relative;
`;

const InputStyled = styled.input`
  margin: 10px 0;
  width: ${props => props.theme.width.formElementRegular};
  height: ${props => props.theme.height.regular};
  padding-left: 50px;
  color: ${props => props.theme.colour.grey};

  @media (max-width: ${breakpoints.sm}) {
    width: ${props => props.theme.width.formElementSmall};
  }

  ::placeholder {
    color: ${props => props.theme.colour.grey};
  }

  &:focus {
    color: ${props => props.theme.colour.grey};
  }
`;

const IconWrapper = styled.div`
  svg {
    fill: ${props => props.theme.colour.lightGrey};
    height: 16px;
    position: absolute;
    top: 50%;
    left: 25px;
    transform: translateY(-50%);
  }
`;

type InputProps = {
  type: string,
  placeholder?: string,
  icon?: ?React.Node,
};

export const Input = ({ icon, ...rest }: InputProps) => (
  <InputWrapper>
    {icon && <IconWrapper>{icon}</IconWrapper>}
    <InputStyled {...rest} className="form-control" />
  </InputWrapper>
);

Input.defaultProps = {
  placeholder: '',
  icon: null,
};
