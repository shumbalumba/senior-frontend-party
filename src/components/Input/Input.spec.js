import * as React from 'react';
import { shallow } from 'enzyme';
import { Input, IconWrapper } from './Input';

describe('<Input />', () => {
  it('renders', () => {
    const wrapper = shallow(<Input type="text" />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Input type="text" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders icon', () => {
    const wrapper = shallow(<Input type="text" />);
    expect(wrapper.find(IconWrapper).exists()).toBeFalsy();
    const fakeIcon = <div>icon</div>;
    wrapper.setProps({
      icon: fakeIcon,
    });
    expect(wrapper.find(IconWrapper).exists()).toBeTruthy();
  });
});
