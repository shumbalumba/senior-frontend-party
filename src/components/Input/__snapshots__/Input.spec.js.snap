// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<styled.div>
  <styled.input
    className="form-control"
    placeholder=""
    type="text"
  />
</styled.div>
`;
